File: src/adf_core_python/core/component/communication/communication_module.py
from __future__ import annotations

from abc import ABC, abstractmethod
from typing import TYPE_CHECKING

if TYPE_CHECKING:
  from adf_core_python.core.agent import Agent
  from adf_core_python.core.agent.communication import MessageManager


class CommunicationModule(ABC):
  @abstractmethod
  def receive(self, agent: Agent, message_manager: MessageManager) -> None:
    pass

  @abstractmethod
  def send(self, agent: Agent, message_manager: MessageManager) -> None:
    pass

File: java/lib/src/main/java/adf_core_python/core/component/communication/CommunicationModule.java
package adf_core_python.core.component.communication;

import adf_core_python.core.agent.Agent;
import adf_core_python.core.agent.communication.MessageManager;

abstract public class CommunicationModule {

    abstract public void receive(Agent agent, MessageManager messageManager);

    abstract public void send(Agent agent, MessageManager messageManager);
}
